chpl/chpl-service/src/main/java/gov/healthit/chpl/upload/listing/ListingUploadHandlerUtil.java

```diff
@@ -0,0 +1,394 @@
+package gov.healthit.chpl.upload.listing;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.validation.ValidationException;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVParser;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.csv.CSVRecord;
+import org.apache.commons.csv.QuoteMode;
+import org.apache.commons.lang.math.IntRange;
+import org.apache.commons.lang.math.Range;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.util.ErrorMessageUtil;
+import lombok.extern.log4j.Log4j2;
+
+@Component("listingUploadHandlerUtil")
+@Log4j2
+public class ListingUploadHandlerUtil {
+    private static final String CRITERION_COL_HEADING_REGEX = "CRITERIA_(\\d+)_(\\d+)_([A-Z])_([0-9]+)([A-Z])?(_CURES)?__C";
+    protected static final String CRITERIA_CURES_COL_HEADING = "CURES";
+    private static final String UPLOAD_DATE_FORMAT = "yyyyMMdd";
+    private DateFormat dateFormat;
+    private Pattern criterionColHeadingPattern;
+    private ErrorMessageUtil msgUtil;
+
+    @Autowired
+    public ListingUploadHandlerUtil(ErrorMessageUtil msgUtil) {
+        this.msgUtil = msgUtil;
+        this.dateFormat = new SimpleDateFormat(UPLOAD_DATE_FORMAT);
+        this.criterionColHeadingPattern = Pattern.compile(CRITERION_COL_HEADING_REGEX);
+    }
+
+    public int getHeadingRecordIndex(List<CSVRecord> allCsvRecords) {
+        int headingIndex = -1;
+        if (allCsvRecords != null) {
+            Optional<CSVRecord> headingRecord =
+                    allCsvRecords.stream().filter(currRecord -> hasHeading(currRecord))
+                    .findFirst();
```

Comment:
Does this need to be a findFirst or can it be a findAny?  It's just a minor performance difference.